### PR TITLE
chore: Add Langchain standard vectorstore tests.

### PR DESCRIPTION
**Pull request by dishaprakash**
Adding Langchain standard test suite for vectorstore for edge cases.

Fixes [b/375659546](https://b.corp.google.com/issues/375659546)

### CODE REVIEW

File: tests/test_standard_test_suite.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            async_engine,
+            embedding_service=self.get_embeddings(),
+            table_name=DEFAULT_TABLE,
+        )
+
+        yield vs
+
+    @property
+    def has_sync(self) -> bool:
+        return False  # Skip sync tests for async vector store
```
Comment (averikitsch) [inline tests/test_standard_test_suite.py:183]: Let's remove the has_sync and has_async for both sets of tests so it will test both the sync and async methods  to ensure our mixed interface works.

**Review by averikitsch — APPROVED** [review]
I've updated this PR to be against a branch "std-tests" so we can collect all the changes before we merge into main.

**Review by averikitsch — CHANGES_REQUESTED** [review]
Looks like the async tests are being skipped
Step #3 - "Run integration tests": tests/test_standard_test_suite.py: 24 warnings
Step #3 - "Run integration tests":   /builder/home/.local/lib/python3.9/site-packages/_pytest/python.py:148: PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.

File: tests/test_standard_test_suite.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            table_name=DEFAULT_TABLE_SYNC,
+        )
+        yield vs
+
+    @property
+    def has_async(self) -> bool:
+        return False  # Skip async tests for sync vector store
+
+
+class TestStandardSuiteAsync(VectorStoreIntegrationTests):
```
Comment (averikitsch) [inline tests/test_standard_test_suite.py:117]: I think we need to add @pytest.mark.asyncio

File: tests/test_standard_test_suite.py (hunk 1 above)
Comment (dishaprakash) [reply tests/test_standard_test_suite.py]: Updated
